react-app/src/app/screens/Game/components/Moves/components/Move/index.js

```diff
@@ -8,21 +8,32 @@ import calculateWinner from '../../../../utils';
 import style from './styles.scss';
 
 class Move extends Component {
+  state = {
+    history: this.props.history,
+    squares: [],
+    winner: null,
+    desc: this.props.move ? `Go to move # ${this.props.move}` : `Go to game start`
+  };
+  static getDerivedStateFromProps(nextProps, prevState) {
+    if (nextProps.history !== prevState.history) {
+      const current = nextProps.history[nextProps.history.length - 1];
+      const currentSquares = current.squares.asMutable().slice();
+      return {
+        history: nextProps.history,
+        squares: currentSquares,
```

Comment:
Notice that history and squares arent being used in this component. Its only needed for calculating the winner and description. They shouldnt be mapped in the local state